logging/logging.go

```diff
@@ -404,7 +404,11 @@ func monitoredResource(parent string) *mrpb.MonitoredResource {
 }
 
 func regionFromZone(zone string) string {
-	return zone[:strings.LastIndex(zone, "-")]
+	cutoff := strings.LastIndex(zone, "-")
+	if cutoff > 0 {
+		return zone[:cutoff]
+	}
+	return zone
```

Comment:
If this value is not what is expected should we put zone in at all? Or just `""`?

Reply:
Maybe I can help out here, in my experience, passing in the values causing this to panic (e.g. `us16`) will work and be accepted by Cloud Logging. I've been using my own autodetect code to stand up our logging client which does this and have had no issues logging:

![image](https://user-images.githubusercontent.com/1321933/116739889-37cf1580-a9c2-11eb-9203-4294106c59b1.png)

Reply:
We should accept any user specified/custom zones. Per user docs: 
`CommonResource sets the monitored resource associated with all log entries written from a Logger. If not provided, the resource is automatically detected based on the running environment (on GCE, GCR, GCF and GAE Standard only). This value can be overridden per-entry by setting an Entry's Resource field.`